test/controllers/CharitableEligibilityControllerSpec.scala

```diff
@@ -16,45 +16,87 @@
 
 package controllers
 
+import controllers.connectors.DataShortCacheConnector
 import helpers.TestHelper
+import models.YesNoModel
+import org.mockito.ArgumentMatchers.any
+import org.mockito.Mockito._
+import org.scalatest.BeforeAndAfterEach
+import play.api.Application
 import play.api.http.Status
+import play.api.inject._
+import play.api.inject.guice.GuiceApplicationBuilder
+import play.api.mvc.{AnyContentAsEmpty, AnyContentAsFormUrlEncoded}
 import play.api.test.FakeRequest
+import uk.gov.hmrc.http.SessionKeys
+import uk.gov.hmrc.http.cache.client.CacheMap
 
+import scala.concurrent.Future
 
+class CharitableEligibilityControllerSpec extends TestHelper with BeforeAndAfterEach {
 
-class CharitableEligibilityControllerSpec extends TestHelper {
+  lazy val mockDataShortCacheConnector: DataShortCacheConnector = mock[DataShortCacheConnector]
 
-  def testController() = new CharitableEligibilityController()(mockAppConfig, mcc)
+  override lazy val fakeRequest: FakeRequest[AnyContentAsEmpty.type] = FakeRequest().withSession(SessionKeys.sessionId -> "foo")
 
-    "CharitableEligibilityController" should {
+  override lazy val fakeApplication: Application = new GuiceApplicationBuilder()
```

Comment:
Hmm, I'm against using fake applications over mocking, was there any reason we made the swap?

Reply:
We are injecting mock of DataCashingConnector to app so easy to handle dependencies.

Reply:
Creating a controller with mocked variables is faster and more efficient for tests. If you're going to pass in a mock anyway then I'm not sure the benefit of the fake application rather than slower tests?